src/FSharp.Stats/Algebra/LinearAlgebra.fs

```diff
@@ -219,6 +219,96 @@ module LinearAlgebra =
         //                    else LinearAlgebraManaged.QR a
         LinearAlgebraManaged.QR a
 
+    /// Performs QR decomposition using an alternative algorithm.
+    /// Returns the orthogonal matrix Q and the upper triangular matrix R.
+    let qrAlternative (A: Matrix<float>) =
+        let m: int = A.NumRows
+        let n: int = A.NumCols
+
+        let q: Matrix<float> = Matrix.zero m n
+        let r: Matrix<float> = Matrix.zero n n
+        let qLengths: Vector<float> = Vector.zeroCreate n
+
+        let getVectorLength (v: Vector<float>) = Vector.fold (fun folder i -> folder+(i*i)) 0. v
+
+        let setqOfA (n: int) =
+            let aN: Vector<float> =  Matrix.getCol A n
+            let qN = 
+                if n = 0 then 
+                    aN 
+                else 
+                    Array.init (n) (fun i -> 
+                        let denominator = qLengths[i]
+                        let forNominator: Vector<float> = Matrix.getCol q i 
+                        let nominator: float = Vector.dot aN forNominator
+                        r.[i, n] <- nominator
+                        (nominator/denominator) * forNominator
+                    )
+                    |> Array.fold (fun folder  e -> folder-e ) aN
+            Matrix.setCol q n qN
+            qN  
+
+        for i=0 to n-1 do
+            let qN = setqOfA i 
+            let qLength = getVectorLength qN
+            let rValue = sqrt(qLength)
+            r[i,i] <- rValue
+            qLengths[i] <- qLength
+
+        for i=0 to n-1 do
+            let qN: Vector<float> = Matrix.getCol q i
+            let updateQ = (1./sqrt( qLengths[i]  )) * qN 
+            Matrix.setCol q i updateQ
+            for j=i+1 to n-1 do
+                let denominator = r[i, i]
+                let nominator = r[i, j]
+                r[i, j] <- (nominator/denominator)
+
+        q, r
+
+    /// Solves a linear system of equations using QR decomposition.
+    /// 
+    /// Parameters:
+    ///   - A: The coefficient matrix of the linear system.
+    ///   - t: The target vector of the linear system.
+    /// 
+    /// Returns:
+    ///   - mX: The solution vector of the linear system.
+    ///   - r: The upper triangular matrix obtained from QR decomposition.
```

Comment:
Thanks for the updates!
I'm going to merge this PR if you just could add proper XML docs like seen in:
https://github.com/fslaborg/FSharp.Stats/blob/c5444cc52c2e58b92d34f10c0c6de90fc4625433/src/FSharp.Stats/Fitting/LinearRegression.fs#L128-L145

The same for line 222 f.